specification/dartLangSpec.tex

```diff
@@ -10372,12 +10374,50 @@ \subsubsection{Binding Actuals to Formals}
 It is a compile-time error if $q_j = q_k$ for any $j \ne k$.
 
 \LMHash{}%
-If the static type of $f$ is \DYNAMIC{} or the built-in class \FUNCTION{},
-no further static checks are performed and the static type of $i$ is \DYNAMIC{};
-otherwise, it is a compile-time error if the static type of $f$ is not a function type.
+For a given type $T_0$, we introduce the notion of a
+\IndexCustom{$T_0$ bounded type}{type!T0 bounded}:
+$T_0$ itself is $T_0$ bounded;
+if $B$ is $T_0$ bounded and
+$X$ is a type variable with bound $B$
+then $X$ is $T_0$ bounded;
+finally, if $B$ is $T_0$ bounded and
+$X$ is a type variable
+then $X \& B$ is $T_0$ bounded.
+In particular, a
+\IndexCustom{\DYNAMIC{} bounded type}{type!dynamic bounded}
+is either \DYNAMIC{} itself
+or a type variable whose bound is \DYNAMIC{} bounded,
+or an intersection whose second operand is \DYNAMIC{} bounded.
+Similarly for a
+\IndexCustom{\FUNCTION{} bounded type}{type!function bounded}.
+
+\LMHash{}%
+Similarly, we introduce the notion of a
+\IndexCustom{function-type bounded type}{type!function-type bounded}:
```

Comment:
How does being "function-type bounded with associated function type T" differ from being a "T bounded type" where T is a function type?

Maybe we can just associate a bound type with every bounded type. If something is `int`-bounded, then its associated bound type is `int`. 
Or is that a problem because then the associated type of `X & Y & T` could be both `T` and `Y & T`?
Can we just say that it's the unique non-intersection, non-type variable type that the type is type-bounded by.

Or we could just say:

> A *function-type bounded type* is a type which is *T* bounded where *T* is a function type.
> A function-type bounded type *T* has an associated function type *F*, 
> which is the unique function type *F* such that the type is an *F* bounded type.

Reply:
Yes, I think that will work. Done.